edc-dataplane/edc-dataplane-proxy-provider-core/src/main/java/org/eclipse/tractusx/edc/dataplane/proxy/provider/core/gateway/auth/RsaPublicKeyParser.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.tractusx.edc.dataplane.proxy.provider.core.gateway.auth;
+
+import org.eclipse.edc.spi.EdcException;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+/**
+ * A thread-safe parser than can read RSA public keys stored using PEM encoding.
+ */
+public class RsaPublicKeyParser {
```

Comment:
this class seems to do the exact same thing as EDC's `PublicKeyParser`, i.e. converting pem-formatted text to a `PublicKey` instance. Could we just use that class instead?

Reply:
so the best course of action would be to a) either make it available through an SPI package, or disallow the usage, even though that means potential code duplication (and version clashes of e.g. nimbus). It needs further discussion in the EDC technical committee.